File: app/src/main/java/farasense/mobile/model/DAO/FaraSenseSensorDailyDAO.java
package farasense.mobile.model.DAO;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

import farasense.mobile.model.DAO.base.BaseDAO;
import farasense.mobile.model.realm.FaraSenseSensorDaily;
import io.realm.Realm;
import io.realm.RealmResults;

public class FaraSenseSensorDailyDAO extends BaseDAO {

    private static final String FIELD_ID = "id";
    private static final String FIELD_TIMESTAMP = "timestamp";
    private static final String FIELD_DATE = "date";

    private static List<FaraSenseSensorDaily> savedFromServer;
    private static List<FaraSenseSensorDaily> dayMeasures;

    public static List<FaraSenseSensorDaily> saveFromServer(final List<FaraSenseSensorDaily> response) {
        Realm realm = null;
        savedFromServer = new ArrayList<>();

        try {
            realm = Realm.getDefaultInstance();
            realm.executeTransaction(realm1 -> {

                if (response != null) {

                    for (FaraSenseSensorDaily faraSenseSensorDaily : response) {
                        faraSenseSensorDaily.setTimestamp(faraSenseSensorDaily.getDate().getTime());
                        savedFromServer.add(faraSenseSensorDaily);
                    }

                    realm1.copyToRealmOrUpdate(savedFromServer);
                } else { savedFromServer = null; }
            });
        } finally {
            if (realm != null) { realm.close(); }
        }

        return savedFromServer;
    }

    public static List<FaraSenseSensorDaily> getMeasureByIntervals(Date startIntervalDay, Date endIntervalDay) {
        Realm realm = null;
        dayMeasures = new ArrayList<>();

        try{
            realm = Realm.getDefaultInstance();
            realm.executeTransaction(new Realm.Transaction() {
                @Override
                public void execute(Realm realm) {
                    RealmResults<FaraSenseSensorDaily> results = realm.where(FaraSenseSensorDaily.class)
                            .between(FIELD_DATE, startIntervalDay, endIntervalDay)
                            .findAll()
                            .sort(FIELD_DATE);

                    if (results.size() != 0) {
                        dayMeasures = realm.copyFromRealm(results);
                    } else {
                        dayMeasures = null;
                    }
                }
            });
        } finally {
            if (realm != null) {
                realm.close();
            }
            return dayMeasures;
        }
    }

}

File: app/src/main/java/old_farasense/model/DAO/FaraSenseSensorDailyDAO.kt
//package old_farasense.model.DAO
//
//import old_farasense.model.DAO.base.BaseDAO
//import old_farasense.model.realm.FaraSenseSensorDaily
//import io.realm.Realm
//import java.util.*
//
//class FaraSenseSensorDailyDAO : BaseDAO() {
//
//    private val FIELD_ID = "id"
//    private val FIELD_TIMESTAMP = "timestamp"
//    private val FIELD_DATE = "date"
//
//    fun saveFromServer(response: List<FaraSenseSensorDaily>?): List<FaraSenseSensorDaily> {
//
//        val savedFromServer: MutableList<FaraSenseSensorDaily> = mutableListOf()
//
//        if (response != null) {
//            Realm.getDefaultInstance().use { realm ->
//
//                realm.executeTransaction{
//
//                    for (faraSenseSensorDaily in response) {
//                        faraSenseSensorDaily.timestamp = faraSenseSensorDaily.date.time
//                        savedFromServer.add(faraSenseSensorDaily)
//                    }
//
//                    it.copyToRealmOrUpdate(savedFromServer)
//                }
//            }
//        }
//
//        return savedFromServer
//    }
//
//    fun getMeasureByIntervals(startIntervalDay: Date, endIntervalDay: Date): List<FaraSenseSensorDaily> {
//        var dayMeasures: MutableIterable<FaraSenseSensorDaily> = mutableListOf()
//
//        Realm.getDefaultInstance().use { realm ->
//            val results = realm.where(FaraSenseSensorDaily::class.java)
//                    .between(FIELD_DATE, startIntervalDay, endIntervalDay)
//                    .findAll()
//                    .sort(FIELD_DATE)
//
//            if (results.size != 0) {
//                dayMeasures = realm.copyFromRealm(results).toMutableList()
//            }
//
//            return dayMeasures.toList()
//        }
//    }
//}
